src/other/index.jl

```diff
@@ -141,7 +141,7 @@ end
 
 function Base.insert!(x::Index, idx::Integer, nm::Symbol)
     if !(1 <= idx <= length(x.names)+1)
-        throw(BoundsError("attempt to insert a column to a data frame with $(length(x)) columns at index $idx"))
+        throw(BoundsError(x, idx))
```

Comment:
Does this print an interpretable message? `x` is just an `Index`, I'm not even sure the message prints its length.

Reply:
No it does not, but `BoundsError` does not allow for printing this information. We could switch `BoundsError` to `ArgumentError`, but recently exactly the opposite change was proposed and implemented as far as I remember. Here I am only cleaning-up non-informative error message we have now.

Another decision could be to revert `BoundsError` to `ArgumentError` or to print `@error` message before throwing an error (this is a bit problematic though if the user has code inside `try`-`catch` and does not want to see these `@error` messages in the terminal).

What do you think would be best?

Reply:
Maybe we could catch the error and rethrow it with more user-friendly information from `getindex(::AbstractDataFrame, ...)`? Or would that require too much code duplication?

Reply:
In will not propagate back to `getindex`. We call it only from `insertcols!`, where we actually have an outside check (so this error in practice should be never triggered) - which I have updated to throw `DimensionMismatch` error which allows an error message (this is not a strictly correct use of `DimensionMismatch` but I felt that it is a less evil).

Reply:
OK, great. Is that the same for the `getindex(x::AbstractIndex, ...)` method below?

Reply:
I will rework everything in this PR to use the `showerror` suggestion and it should not matter then.